packages/vite/src/node/plugins/html.ts

```diff
@@ -323,6 +323,22 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
             config,
             publicToRelative,
           )
+        const nodeStartWithLeadingWhitespace = (
+          node: DefaultTreeAdapterMap['node'],
+        ) => {
+          if (node.sourceCodeLocation!.startOffset == 0)
+            return node.sourceCodeLocation!.startOffset
```

Comment:
This should exist solely to satisfy the test suite, as I hope no one is actually writing HTML docs that immediately start with a `<script>` or `<link>` (creating a `startOffset` of `0` w/ a `startCol` of `1`)